lib/sync/index.js

```diff
@@ -121,24 +123,52 @@ function syncRecords(dataset_id, params, cb) {
   return cb(null, {});
 }
 
+
+/**
+ * Connect to mongodb & redis with the given connection urls.
+ * This should be called before `start()` is called
+ * 
+ * @param {string} mongoDBConnectionUrl A MongoDB connection uril in a format supported by the `mongodb` module
+ * @param {string} redisUrl A redis connection url in a format supported by the `redis` module
+ * @param {function} cb 
+ */
 function connect(mongoDBConnectionUrl, redisUrl, cb) {
-  async.parallel([
+  if (arguments.length < 3) throw new Error('connect requires 3 arguments');
+  
+  async.series([
     function connectToMongoDB(callback) {
       MongoClient.connect(mongoDBConnectionUrl, callback);
     },
     function connectToRedis(callback) {
-      var client = redis.createClient(redisUrl);
+      var redisOpts = {
+        url: redisUrl
+      };
+      var client = redis.createClient(redisOpts);
       return callback(null, client);
     }
   ], function(err, results) {
     if (err) return cb(err);
     mongoDbClient = results[0];
     redisClient = results[1];
-    return cb();
+    return cb(null, mongoDbClient, redisClient);
   });
 }
 
+/**
+ * Starts all sync queues, workers & the sync scheduler.
+ * This should only be called after `connect()`.
+ * If this is not explicitly called before clients send sync requests,
+ * it will be called when the first client sends a sync request
+ * 
```

Comment:
in this case, we can added a in memory flag to return early if `start` is already being called.